ivre/db/mongo.py

```diff
@@ -721,6 +721,52 @@ def getscreenshot(port):
         if url == "field":
             return port.get('screendata')
 
+    def set_label(self, flt, group, label, archives=False):
+        """Adds `label` in `group` to every host matching `flt`"""
+        colname = self.colname_oldhosts if archives else self.colname_hosts
+        for host in self.find(colname, flt):
+            labels = host.setdefault('labels', [])
+            try:
+                g_label = (lab for lab in labels if lab['group'] == group).next()
```

Comment:
You should only catch the exception for this line (and use an `else:` block for the remaining code).